src/Gui/ElideLabel.cpp

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2025 Alfredo Monclus <alfredomonclus@gmail.com>          *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+// This custom widget adds the missing ellipsize functionality in QT5
+
+#include "PreCompiled.h"
+
+#include "ElideLabel.h"
+
+ElideLabel::ElideLabel(QWidget *parent)
+    : QLabel(parent) {
+}
+
+void ElideLabel::paintEvent(QPaintEvent *event) {
+    Q_UNUSED(event);
+    QPainter painter(this);
+    painter.setPen(palette().color(QPalette::WindowText));
+    painter.setFont(font());
+
+    QFontMetrics fm(font());
+    QString text = this->text();
+    int availableWidth = width() - 4;  // Account for padding
```

Comment:
Can we move the padding to static constexpr? then 4 and 2 on 46 line could be replaced with expressions like `width() - padding` and `padding / 2`